File: back-end/DAL_01/BOL_02/CarModel.cs
using System.ComponentModel.DataAnnotations;

namespace BOL_02
{
    public class CarModel
    {
        public int CarID { get; set; }


        public int CarTypeID { get; set; }

        [Required, RegularExpression("^[0-9]*$", ErrorMessage = "characters and special characters are not allowed in the name Model field")]
        public int CurrentKilometrage { get; set; }

        [Required]
        public string Image { get; set; }

        [Required]
        public bool IsProperToRent { get; set; }

        [Required, RegularExpression("^[0-9]*$", ErrorMessage = "characters and special characters are not allowed in the name Model field")]
        public int CarNumber { get; set; }

        //[Required, RegularExpression("^[0-9]*$", ErrorMessage = "characters and special characters are not allowed in the name Model field")]
        public int LocatedAtTheBranch { get; set; }


    }
}

File: front-end/project-car-rent/src/app/shared/models/car.model.ts
export class Car {
public CarID: number;
public CarTypeID: number;
public CurrentKilometrage: number;
public Image: string;
public IsProperToRent: boolean;
public CarNumber: number;
public LocatedAtTheBranch: number;



constructor(CarID ?: number,
    CarTypeID ?: number,
    CurrentKilometrage?: number,
    Image: string = '',
    IsProperToRent?: boolean,
    CarNumber?: number,
    LocatedAtTheBranch?: number) {
        this.CarID = CarID;
            this.CarTypeID = CarTypeID ;
            this.CurrentKilometrage = CurrentKilometrage ;
            this.Image = Image ;
            this.IsProperToRent = IsProperToRent ;
            this.CarNumber = CarNumber ;
            this.LocatedAtTheBranch = LocatedAtTheBranch;
        }
    }
